dotCMS/src/main/java/com/dotcms/rendering/velocity/util/VelocityUtil.java

```diff
@@ -0,0 +1,607 @@
+		context.put("velocityContext", context);
+		context.put("language", "1");
+		context.put("InodeUtils", new InodeUtils());
+		return context;
+	}
+	
+	
+	/**
+	 * Gets creates Velocity context will all the toolbox, user, host, language and request stuff
+	 * inside the map
+	 * @param request
+	 * @param response
+	 * @return
+	 */
+	public static ChainedContext getWebContext(HttpServletRequest request, HttpServletResponse response) {
+		return getWebContext(getBasicContext(), request, response);
+	}
+	
+	
+	public static ChainedContext getWebContext(Context ctx, HttpServletRequest request, HttpServletResponse response) {
+
+        if ( ctx == null ) {
+            ctx = getBasicContext();
+        }
+
+        // http://jira.dotmarketing.net/browse/DOTCMS-2917
+
+		//get the context from the request if possible
+        ChainedContext context;
+        if ( request.getAttribute( com.dotcms.rendering.velocity.Constants.VELOCITY_CONTEXT ) != null && request.getAttribute( com.dotcms.rendering.velocity.Constants.VELOCITY_CONTEXT ) instanceof ChainedContext ) {
+            return (ChainedContext) request.getAttribute( "velocityContext" );
+        } else {
+            RequestWrapper rw = new RequestWrapper( request );
+            if ( request.getAttribute( "User-Agent" ) != null && request.getAttribute( "User-Agent" ).equals( Constants.USER_AGENT_DOTCMS_BROWSER ) ) {
+                rw.setCustomUserAgentHeader( Constants.USER_AGENT_DOTCMS_BROWSER );
+            }
+            context = new ChainedContext( ctx, getEngine(), rw, response, Config.CONTEXT );
+        }
+
+        context.put("context", context);
+		Logger.debug(VelocityUtil.class, "ChainedContext=" + context);
+		/*
+		 * if we have a toolbox manager, get a toolbox from it See
+		 * /WEB-INF/toolbox.xml
+		 */
+		context.setToolbox(getToolboxManager().getToolboxContext(context));
+
+
+		// put the list of languages on the page
+		context.put("languages", getLanguages());
+		HttpSession session = request.getSession(false);
+		if(!UtilMethods.isSet(request.getAttribute(WebKeys.HTMLPAGE_LANGUAGE)) && session!=null)
+		    context.put("language", (String) session.getAttribute(com.dotmarketing.util.WebKeys.HTMLPAGE_LANGUAGE));
+		else
+		    context.put("language", request.getAttribute(WebKeys.HTMLPAGE_LANGUAGE));
+
+		try {
+			Host host;
+			host = WebAPILocator.getHostWebAPI().getCurrentHost(request);
+			context.put("host", host);
+		} catch (Exception e) {
+			Logger.error(VelocityUtil.class,e.getMessage(),e);
+		}
+		context.put("pdfExport", false);
+        context.put("dotPageMode", PageMode.get(request));
+		if(request.getSession(false)!=null){
+			try {
+				User user = (com.liferay.portal.model.User) request.getSession().getAttribute(com.dotmarketing.util.WebKeys.CMS_USER);
+				context.put("user", user);
+
+				Visitor visitor = (Visitor) request.getSession().getAttribute(WebKeys.VISITOR);
+				context.put("visitor", visitor);
+
+			} catch (Exception nsue) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)